File: src/backend/src/models/homebrew.py
from pydantic import BaseModel
from typing import List, Literal, Optional


class HomebrewDocumentSummary(BaseModel):
    """Summary model for listing homebrew documents"""
    id: str
    title: str


class HomebrewDocument(BaseModel):
    """Model representing a full homebrew document with content"""
    id: str
    title: str
    content: str


class HomebrewTreeNode(BaseModel):
    """A node in the homebrew file tree. Can be a file or a directory."""
    name: str
    type: Literal["file", "directory"]
    path: str
    children: Optional[List["HomebrewTreeNode"]] = None

File: src/frontend/src/types/homebrew.ts
export interface HomebrewDocumentSummary {
  id: string;
  title: string;
}

export interface HomebrewDocument {
  id: string;
  title: string;
  content: string;
}

export interface HomebrewTreeNode {
  name: string;
  type: 'file' | 'directory';
  path: string;
  children?: HomebrewTreeNode[];
}
